File: web_manager/src/main/java/cn/itcast/core/controller/SeckillOrderController.java
package cn.itcast.core.controller;


import cn.itcast.core.pojo.seckill.SeckillOrder;
import cn.itcast.core.service.SeckillOrderService;
import com.alibaba.dubbo.config.annotation.Reference;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;

@RestController
@RequestMapping("/seckillOrderService")
public class SeckillOrderController {

    @Reference
    private SeckillOrderService seckillOrderService;

    @RequestMapping("/findAll")
    public List<SeckillOrder>findAll(){
        return seckillOrderService.findAll();
    }


    @RequestMapping("/search")
    public List<SeckillOrder> search(@RequestBody(required = false)SeckillOrder seckillOrder){
        return seckillOrderService.search(seckillOrder);
    }
}

File: web_manager/src/main/webapp/js/controller/seckillOrderController.js
//控制层
app.controller('seckillOrderController' ,function($scope,$controller,$location,seckillOrderService){

    $controller('baseController',{$scope:$scope});//继承

    //读取列表数据绑定到表单中
    $scope.findAll=function(){
        seckillOrderService.findAll().success(
            function(response){
                $scope.list=response;
            }
        );
    }


    $scope.searchEntity={};//定义搜索对象

    //搜索
    $scope.search=function(){
        seckillOrderService.search($scope.searchEntity).success(
            function(response){
                $scope.list=response;

            }
        );
    }
});

